File: app/src/main/java/eu/warble/pjapp/util/Converter.java
package eu.warble.pjapp.util;


import android.support.annotation.NonNull;

import com.google.gson.Gson;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import eu.warble.pjapp.data.model.Student;
import okhttp3.Response;
import okhttp3.ResponseBody;

public class Converter {

    public static String responseToJsonString(@NonNull ResponseBody body) throws IOException{
        StringBuilder builder = new StringBuilder();
        String tmp;
        try(BufferedReader br = new BufferedReader(new InputStreamReader(body.byteStream()))) {
            while ((tmp = br.readLine()) != null)
                builder.append(tmp);
        }
        return builder.toString();
    }

    public static Student jsonStringToStudentData(String json){
        if (json != null)
            return new Gson().fromJson(json, Student.class);
        return null;
    }

    public static String studentDataToJsonString(Student studentData){
        if (studentData != null)
            return new Gson().toJson(studentData);
        return null;
    }

}

File: app/src/main/java/eu/warble/pjappkotlin/utils/Converter.kt
package eu.warble.pjappkotlin.utils

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import eu.warble.pjappkotlin.data.model.NewsItem
import eu.warble.pjappkotlin.data.model.Student
import eu.warble.pjappkotlin.data.model.ZajeciaItem
import okhttp3.ResponseBody
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

object Converter {

    @Throws(IOException::class)
    fun responseToString(body: ResponseBody): String {
        val builder = StringBuilder()
        BufferedReader(InputStreamReader(body.byteStream())).use { br ->
            var tmp: String? = br.readLine()
            while (tmp != null) {
                builder.append(tmp)
                tmp = br.readLine()
            }
        }
        return builder.toString()
    }

    fun jsonStringToScheduleList(json: String?): List<ZajeciaItem>? {
        if (json != null) {
            val listType = object : TypeToken<List<ZajeciaItem>>() {}.type
            return Gson().fromJson(json, listType)
        }
        return null
    }

    fun jsonStringToNewsList(json: String?): ArrayList<NewsItem>? {
        if (json != null) {
            val listType = object : TypeToken<ArrayList<NewsItem>>() {}.type
            return Gson().fromJson(json, listType)
        }
        return null
    }

    fun scheduleListToJsonString(data: List<ZajeciaItem>?): String? {
        return if (data != null) Gson().toJson(data) else null
    }

    fun jsonStringToStudentData(json: String?): Student? {
        if (json != null) {
            return Gson().fromJson(json, Student::class.java)
        }
        return null
    }

    fun studentDataToJsonString(studentData: Student?): String? {
        return if (studentData != null) Gson().toJson(studentData) else null
    }
}